testing/mysql/integration_test.go

```diff
@@ -0,0 +1,203 @@
+	"cloud.google.com/go/spanner"
+	database "cloud.google.com/go/spanner/admin/database/apiv1"
+	instance "cloud.google.com/go/spanner/admin/instance/apiv1"
+	"google.golang.org/api/iterator"
+
+	databasepb "google.golang.org/genproto/googleapis/spanner/admin/database/v1"
+)
+
+var (
+	projectID  string
+	instanceID string
+
+	instanceAdmin *instance.InstanceAdminClient
+	databaseAdmin *database.DatabaseAdminClient
+)
+
+func TestMain(m *testing.M) {
+	cleanup := initIntegrationTests()
+	res := m.Run()
+	cleanup()
+	os.Exit(res)
+}
+
+func initIntegrationTests() (cleanup func()) {
+	projectID = os.Getenv("HARBOURBRIDGE_TESTS_GCLOUD_PROJECT_ID")
+	instanceID = os.Getenv("HARBOURBRIDGE_TESTS_GCLOUD_INSTANCE_ID")
+
+	ctx := context.Background()
+	flag.Parse() // Needed for testing.Short().
+	noop := func() {}
+
+	if testing.Short() {
+		log.Println("Integration tests skipped in -short mode.")
+		return noop
+	}
+
+	if projectID == "" {
+		log.Println("Integration tests skipped: HARBOURBRIDGE_TESTS_GCLOUD_PROJECT_ID is missing")
+		return noop
+	}
+
+	if instanceID == "" {
+		log.Println("Integration tests skipped: HARBOURBRIDGE_TESTS_GCLOUD_INSTANCE_ID is missing")
+		return noop
+	}
+
+	var err error
+	databaseAdmin, err = database.NewDatabaseAdminClient(ctx)
+	if err != nil {
+		log.Fatalf("cannot create databaseAdmin client: %v", err)
+	}
+
+	return func() {
+		databaseAdmin.Close()
+	}
+}
+
+func dropDatabase(t *testing.T, dbPath string) {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+	defer cancel()
+	// Drop the testing database.
+	if err := databaseAdmin.DropDatabase(ctx, &databasepb.DropDatabaseRequest{Database: dbPath}); err != nil {
+		t.Fatalf("failed to drop testing database %v: %v", dbPath, err)
+	}
+}
+
+func cleanupFiles(t *testing.T, files []string) {
+	for _, file := range files {
+		if _, err := os.Stat(file); err == nil {
+			err = os.Remove(file)
+			if err != nil {
+				t.Errorf("failed to delete file: %v", file)
+			}
+		}
+	}
+}
+
+func prepareIntegrationTest(t *testing.T) string {
+	if databaseAdmin == nil {
+		t.Skip("Integration tests skipped")
+	}
+	tmpdir, err := ioutil.TempDir(".", "int-test-")
+	if err != nil {
+		log.Fatal(err)
+	}
+	return tmpdir
+}
+
+func TestIntegration_MYSQLDUMP_SimpleUse(t *testing.T) {
+	t.Parallel()
+
+	tmpdir := prepareIntegrationTest(t)
+	defer os.RemoveAll(tmpdir)
+
+	now := time.Now()
+	dbName, _ := conversion.GetDatabaseName(conversion.MYSQLDUMP, now)
+	dbPath := fmt.Sprintf("projects/%s/instances/%s/databases/%s", projectID, instanceID, dbName)
+	dataFilepath := "../../test_data/mysqldump.test.out"
+	filePrefix := filepath.Join(tmpdir, dbName+".")
+	f, err := os.Open(dataFilepath)
+	if err != nil {
+		t.Fatalf("failed to open the test data file: %v", err)
+	}
+	err = cmd.CommandLine(conversion.MYSQLDUMP, projectID, instanceID, dbName /*dataOnly=*/, false /*schemaOnly=*/, false /*sessionJSON=*/, "", &conversion.IOStreams{In: f, Out: os.Stdout}, filePrefix, now)
```

Comment:
I like these comments, but shouldn't they come after the comma i.e ", /*dataOnly=*/ false"

Reply:
Seems to be an issue with gofmt that I am using in my local editor. What's more is if I do fix these argument comments e.g., `, dbName, /*dataOnly=*/false, ` etc then our lint integration starts complaining due to `goimports` library also expecting the wrong formatting. Have fixed these manually while I look for how to fix these.

Reply:
Seems to be an issue with auto formatting done by gofmt. What's more is that if I do fix these manually, then our lint workflow starts failing with goimports command also expecting this wrong output. Will investigate this more, and fix the lint workflow.

Reply:
Done @https://github.com/cloudspannerecosystem/harbourbridge/pull/121/commits/96fafc8df69686a5aeac12ee5bec2c5656666d1e